src/groqStore.ts

```diff
@@ -20,9 +23,20 @@ export function groqStore(config: Config, envImplementations: EnvImplementations
         (docs) => {
           documents = docs
           executeThrottled()
+          eventEmitter.emit('datasetChanged', {
+            dataset: config.dataset,
+            documents: docs,
+          })
```

Comment:
I think this emitter should be throttled, the same way `executeAllSubscriptions` is.
To avoid a mismatch where you will see `datasetChanged` events fire, but `store.query` and such not yet "see" the changes yet since these are throttled.